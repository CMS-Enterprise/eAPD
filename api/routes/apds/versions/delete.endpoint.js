const {
  getDB,
  getFullPath,
  login,
  request,
  unauthenticatedTest,
  unauthorizedTest
} = require('../../../utils.endpoint');

const url = id => getFullPath(`/apds/${id}/versions`);

describe('apd version withdraw endpoint | DELETE /apds/:id/versions', async () => {
  const db = getDB();
  beforeAll(() => db.seed.run());
  afterAll(() => db.destroy());

  unauthenticatedTest('delete', url(1));
  unauthorizedTest('delete', url(1));

  it('when attempting to withdraw an APD', async () => {
    const jar = await login();
    const json = true;

    // Change the APD's status before attempting to withdraw
    // so we can make sure it is changed back.
    await db('apds')
      .where({ id: 4000 })
      .update({ status: 'not draft' });

    const { response, body } = await request.delete(url(4000), { jar, json });
    const apd = await db('apds')
      .where({ id: 4000 })
      .select();

    expect(response.statusCode).toEqual(204);
    expect(body).toMatchSnapshot();
    expect(apd[0].status).toEqual('draft');
  });
});
