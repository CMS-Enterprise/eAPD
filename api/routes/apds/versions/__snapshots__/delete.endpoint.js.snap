// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd version withdraw endpoint | DELETE /apds/:id/versions when attempting to withdraw an APD 1`] = `undefined`;
